server/routes/createGoal/createGoalsController.ts

```diff
@@ -71,16 +76,23 @@ export default class CreateGoalsController {
       return res.redirectWithErrors(`/plan/${prisonNumber}/goals/create`, errors)
     }
 
+    req.session.createGoalsForm = undefined
+    const createGoalDtos = toCreateGoalDtos(createGoalsForm, prisonId)
+
     try {
-      const createGoalDtos = toCreateGoalDtos(createGoalsForm, prisonId)
       await this.educationAndWorkPlanService.createGoals(prisonNumber, createGoalDtos, req.user.token)
-
-      req.session.createGoalsForm = undefined
-      return res.redirectWithSuccess(`/plan/${prisonNumber}/view/overview`, 'Goals added')
     } catch (e) {
       logger.error(`Error creating goal(s) for prisoner ${prisonNumber}`, e)
       return next(createError(500, `Error creating goal(s) for prisoner ${prisonNumber}. Error: ${e}`))
     }
+
+    Promise.all(
```

Comment:
Does it still need to be wrapped in Promise.all?

Reply:
Good shout, probably not .... let me try ... 😁